addons/l10n_mx/controllers/portal.py

```diff
@@ -0,0 +1,31 @@
+from itertools import count
+
+from odoo.addons.portal.controllers import portal
+from odoo.http import request
+
+class CustomerPortal(portal.CustomerPortal):
+
+    def _get_mandatory_fields(self):
+
+        country_id = request.env.context.get('portal_form_country_id')
+        if country_id:
+            try:
+                country_id = int(country_id)
+            except ValueError:
+                country_id = None
```

Comment:
```suggestion
        try:
            country_id = int(request.env.context.get('portal_form_country_id', ''))
        except ValueError:
            country_id = None
```
The `country_id` var will be initialized on both case.
Adding `''` default value from the `get` so that it will raise ValueError if not found. (`int(None)` -> `TypeError`, `int('')` -> `ValueError`)